tests/unit/mixins/adapter-fetch-test.js

```diff
@@ -172,3 +176,43 @@ test('serialiazeQueryParams turns deeply nested objects into queryParams like $.
 
   assert.equal(queryParamString, 'a=1&b=2&c%5Bd%5D=3&c%5Be%5D%5Bf%5D=4&c%5Bg%5D%5B%5D=5&c%5Bg%5D%5B%5D=6&c%5Bg%5D%5B%5D=7');
 });
+
+test('determineBodyResponse returns the body when it is present', function(assert) {
+  assert.expect(1);
+
+  const done = assert.async();
+  const response = new Response('{"data": "foo"}',
+    {
+      status: 200,
+      headers: new Headers({
+        'Content-Length': 15
+      })
+    }
+  );
+  const bodyPromise = determineBodyPromise(response);
+
+  bodyPromise.then((body) => {
+    assert.deepEqual(body, {data: 'foo'});
+    done();
+  });
+});
+
+test('determineBodyResponse returns an empty object when the body is not present', function(assert) {
+  assert.expect(1);
+
+  const done = assert.async();
+  const response = new Response('',
+    {
+      status: 200,
+      headers: new Headers({
+        'Content-Length': 0
+      })
+    }
+  );
+  const bodyPromise = determineBodyPromise(response);
+
+  bodyPromise.then((body) => {
```

Comment:
rather then `done` lets use: `return bodyPromise.then` That way rejections become quick test failures, rather then timeouts.